public/framework/catalog_cache/cache_manager.test.tsx

```diff
@@ -396,4 +391,84 @@ describe('CatalogCacheManager', () => {
       expect(localStorage.removeItem).toHaveBeenCalledWith(ASYNC_QUERY_ACCELERATIONS_CACHE);
     });
   });
+
+  describe('addOrUpdateAccelerationsByDataSource', () => {
+    it('should add a new data source to the accelerations cache', () => {
+      const dataSource: CachedAcclerationByDataSource = {
+        name: 'TestDataSource',
+        lastUpdated: '2024-03-08T12:00:00Z',
+        status: CachedDataSourceStatus.Updated,
+        accelerations: [],
+      };
+
+      CatalogCacheManager.addOrUpdateAccelerationsByDataSource(dataSource);
+
+      // Verify that saveAccelerationsCache is called with the updated cache data
+      expect(localStorage.setItem).toHaveBeenCalledWith(
+        ASYNC_QUERY_ACCELERATIONS_CACHE,
+        JSON.stringify({
+          version: '1.0',
+          dataSources: [{ ...dataSource }],
+        })
+      );
+    });
+
+    it('should update an existing data source in the accelerations cache', () => {
+      // Set up initial cache data
+      const initialDataSource: CachedAcclerationByDataSource = {
+        name: 'TestDataSource',
+        lastUpdated: '2024-03-08T12:00:00Z',
+        status: CachedDataSourceStatus.Updated,
+        accelerations: [],
+      };
+      // mockAccelerationsCacheData.dataSources = [initialDataSource];
```

Comment:
If this line isn't needed could you remove it?

Reply:
updated PR